lib/templates.php

```diff
@@ -75,19 +75,37 @@ function gutenberg_register_template_post_type() {
 	);
 
 	register_post_type( 'wp_template', $args );
+}
+add_action( 'init', 'gutenberg_register_template_post_type' );
 
-	$meta_args = array(
-		'object_subtype' => 'wp_template',
-		'type'           => 'string',
-		'description'    => 'The theme that provided the template, if any.',
-		'single'         => true,
-		'show_in_rest'   => true,
-	);
+/**
+ * Registers block editor 'wp_theme' taxonomy.
+ */
+function gutenberg_register_wp_theme_taxonomy() {
+	if ( ! gutenberg_is_fse_theme() ) {
+		return;
+	}
 
-	register_post_type( 'wp_template', $args );
-	register_meta( 'post', 'theme', $meta_args );
+	register_taxonomy(
+		'wp_theme',
+		array( 'wp_template', 'wp_template_part' ),
```

Comment:
I find the fact that we are registering in `wp_template` a taxonomy for both `wp_template` and `wp_template_part` slightly confusing.
I don't have a better solution though, unless we created a new file to contain FSE-related taxonomies, and load it after both templates and template parts have been registered.

I don't think it's a blocker either way, just something we might need to remember while working on this stuff. 🤔